### PR TITLE
Add test cases for Various RuleConfigurationToDistSQLConverters

### LINKED ISSUE #30628: Add test cases for implementations of RuleConfigurationToDistSQLConverter
FYI: BroadcastRuleConfigurationToDistSQLConverterTest

### Tasks
- [ ] EncryptRuleConfigurationToDistSQLConverter
- [ ] MaskRuleConfigurationToDistSQLConverter
- [ ] ReadwriteSplittingRuleConfigurationToDistSQLConverter
- [ ] ShadowRuleConfigurationToDistSQLConverter
- [ ] ShardingRuleConfigurationToDistSQLConverter
- [ ] SingleRuleConfigurationToDistSQLConverter

### PR DESCRIPTION
**Pull request by sroopsai**
Fixes #30628.

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Review by linghengqian — COMMENTED** [review]
- It just looks like you rebase wrong, https://github.com/apache/shardingsphere/pull/30658/commits/ef1cff35da3c7c528bcad6ec7ec7b2f17f24a94b should be at the bottom instead of the top.

**Review by linghengqian — COMMENTED** [review]
- The master branch has been updated to a different commit, so you may need to extract your changes to rebase.

**Review by linghengqian — APPROVED** [review]
- LGTM! 

- I am not the author of the original issue, Call @RaigorJiang to help review.

**Comment by sroopsai** [comment]
@RaigorJiang can you review my PR.

File: features/encrypt/distsql/handler/src/test/java/org/apache/shardingsphere/encrypt/distsql/handler/provider/EncryptRuleConfigurationToDistSQLConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+class EncryptRuleConfigurationToDistSQLConverterTest {
+    
+    @Test
+    void assertConvertWithEmptyTables() {
+        EncryptRuleConfiguration encryptRuleConfiguration = mock(EncryptRuleConfiguration.class);
+        when(encryptRuleConfiguration.getTables()).thenReturn(Collections.emptyList());
+        EncryptRuleConfigurationToDistSQLConverter encryptRuleConfigurationToDistSQLConverter = new EncryptRuleConfigurationToDistSQLConverter();
+        assertThat(encryptRuleConfigurationToDistSQLConverter.convert(encryptRuleConfiguration), is(""));
+
```
Comment (RaigorJiang) [inline features/encrypt/distsql/handler/src/test/java/org/apache/shardingsphere/encrypt/distsql/handler/provider/EncryptRuleConfigurationToDistSQLConverterTest.java:45]: Please remove useless empty line.

File: features/encrypt/distsql/handler/src/test/java/org/apache/shardingsphere/encrypt/distsql/handler/provider/EncryptRuleConfigurationToDistSQLConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    
+    @Test
+    void assertConvert() {
+        EncryptRuleConfiguration encryptRuleConfiguration = getEncryptRuleConfiguration();
+        EncryptRuleConfigurationToDistSQLConverter encryptRuleConfigurationToDistSQLConverter = new EncryptRuleConfigurationToDistSQLConverter();
+        assertThat(encryptRuleConfigurationToDistSQLConverter.convert(encryptRuleConfiguration), is("CREATE ENCRYPT RULE t_encrypt (\nCOLUMNS(\n"
+                +
+                "(NAME=user_id, CIPHER=user_cipher, ASSISTED_QUERY_COLUMN=user_assisted, LIKE_QUERY_COLUMN=user_like, ENCRYPT_ALGORITHM(TYPE(NAME='md5')), "
+                +
+                "ASSISTED_QUERY_ALGORITHM(), LIKE_QUERY_ALGORITHM()),\n"
```
Comment (RaigorJiang) [inline features/encrypt/distsql/handler/src/test/java/org/apache/shardingsphere/encrypt/distsql/handler/provider/EncryptRuleConfigurationToDistSQLConverterTest.java:56]: `\n` may cause compatibility issues. Can it be replaced by `System.lineSeparator()`?

File: features/encrypt/distsql/handler/src/test/java/org/apache/shardingsphere/encrypt/distsql/handler/provider/EncryptRuleConfigurationToDistSQLConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                "ASSISTED_QUERY_ALGORITHM(), LIKE_QUERY_ALGORITHM())\n"
+                +
+                ")),\n t_encrypt_another (\nCOLUMNS(\n"
+                +
+                "(NAME=user_id, CIPHER=user_cipher, ASSISTED_QUERY_COLUMN=user_assisted, LIKE_QUERY_COLUMN=user_like, ENCRYPT_ALGORITHM(TYPE(NAME='md5')), "
+                +
+                "ASSISTED_QUERY_ALGORITHM(), LIKE_QUERY_ALGORITHM())\n"
+                +
+                "));"));
+
```
Comment (RaigorJiang) [inline features/encrypt/distsql/handler/src/test/java/org/apache/shardingsphere/encrypt/distsql/handler/provider/EncryptRuleConfigurationToDistSQLConverterTest.java:69]: Please remove useless empty line.

File: features/mask/distsql/handler/src/test/java/org/apache/shardingsphere/mask/distsql/handler/provider/MaskRuleConfigurationToDistSQLConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        assertThat(maskRuleConfigurationToDistSQLConverter.getType().getName(), is("org.apache.shardingsphere.mask.api.config.MaskRuleConfiguration"));
+    }
+    
+    private MaskRuleConfiguration getMaskRuleConfiguration() {
+        MaskColumnRuleConfiguration maskColumnRuleConfig = new MaskColumnRuleConfiguration("user_id", "t_mask_user_id_md5");
+        MaskTableRuleConfiguration maskTableRuleConfig = new MaskTableRuleConfiguration("t_mask", Collections.singleton(maskColumnRuleConfig));
+        AlgorithmConfiguration algorithmConfig = new AlgorithmConfiguration("md5", new Properties());
+        return new MaskRuleConfiguration(Collections.singleton(maskTableRuleConfig), Collections.singletonMap("t_mask_user_id_md5", algorithmConfig));
+    }
+
```
Comment (RaigorJiang) [inline features/mask/distsql/handler/src/test/java/org/apache/shardingsphere/mask/distsql/handler/provider/MaskRuleConfigurationToDistSQLConverterTest.java:63]: Please remove useless empty line.

File: kernel/single/distsql/handler/src/test/java/org/apache/shardingsphere/single/distsql/handler/provider/SingleRuleConfigurationToDistSQLConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        SingleRuleConfigurationToDistSQLConverter singleRuleConfigurationToDistSQLConverter = new SingleRuleConfigurationToDistSQLConverter();
+        assertThat(singleRuleConfigurationToDistSQLConverter.convert(singleRuleConfiguration), is(""));
+    }
+    
+    @Test
+    void assertGetType() {
+        SingleRuleConfigurationToDistSQLConverter singleRuleConfigurationToDistSQLConverter = new SingleRuleConfigurationToDistSQLConverter();
+        assertThat(singleRuleConfigurationToDistSQLConverter.getType().getName(), is("org.apache.shardingsphere.single.api.config.SingleRuleConfiguration"));
+    }
+
```
Comment (RaigorJiang) [inline kernel/single/distsql/handler/src/test/java/org/apache/shardingsphere/single/distsql/handler/provider/SingleRuleConfigurationToDistSQLConverterTest.java:56]: Empty line.

**Review by RaigorJiang — CHANGES_REQUESTED** [review]
Hi @sroopsai , Thanks for your great PR, I found some problems, please fix them.

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
@sroopsai Merged, thank you!

**Pull request merged**